zanata-war/src/main/java/org/zanata/action/SupportedLanguageAction.java

```diff
@@ -41,10 +46,19 @@
     protected List<HLocale> fetchAll() {
         if (supportedLanguages == null) {
             supportedLanguages = localeServiceImpl.getSupportedLocales();
```

Comment:
I have been seeing this pattern a lot lately. We should probably wrap it up in something (not in this PR though).

```
class LazyField<T> {
    private T value = null;

    /**
     * Override this with code to look up the value.
     */
    protected abstract T fetchValue();

    public T get() {
        if (value == null) {
            value = fetchValue();
        }
        return value;
    }
}
```